employee-rostering-gwtui/src/main/resources/org/optaweb/employeerostering/gwtui/client/resources/i18n/OptaWebUIConstants_en.properties

```diff
@@ -43,6 +43,8 @@ Indictment.softScore=Soft Score: {0}
 KiePager.of=of
 KieSearchBar.search=Search
 LoadingSpinner.loading=Loading...
+LocalDate.format=EEE, MMM d, yyyy
+LocalTime.format=h:mm a
```

Comment:
It's a bit strange to see these in i18n.

There are 2 concepts at play:

- Normally date/time patterns are (at least partially) i18n independent. For example, the month resolves to "Oct" in English, but "Okt" in Dutch. That's not a problem.

- Each locale has their own "long date" and "short date" formats. For example, a long format in US would be "October 8th", but in Dutch it's "8 Oktober". Notice that the 8 switches place based on the locale. Now, you might presume that's therefore it needs to be in the properties file. But if you look into the Fedora configuration settings, you'll find that "formats" is separate from "language" (although it is defaulted through the locale). And formats include information such as date order (9/11 vs 11/9), number format (1,000.00 vs 1 000.00), etc. I would presume that in a HTTP request, which specifies the locale - it either specifies the formats too or they get defaulted by locale [1].

What's my point? We just want to choose whether it's a "long" format or "short" format (so "Oct" vs "October" etc), the other stuff (day before month or the reverse) needs to come from the locale - normally the timedate stuff does this for is. If this is all true, we can hard code the date format in the string parameter to the time date formatter.

Note: all calls to the time date formatter must pass in the http request locale (!= server locale) as a parameter.

Reply:
@ge0ffrey GWT does not run an JVM, nor does it implement `java.time` packages. The `java.time` files we use are super-sourced, and a bit outdated. It can only format date to ISO format. GWT provides `DateTimeFormat` to format dates, although none of `DATE_FULL`, `DATE_LONG`, `DATE_MEDIUM` and `DATE_SHORT` (the ones that use locale positioning) have weekday first (as required by https://issues.jboss.org/browse/PLANNER-1097?focusedCommentId=13604997&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13604997). Another possible solution is to use Javascript date: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

Reply:
Ah yes, this stuff is client-side and we don't have full java.time support there. Ok, to leave as is.